[AG-13490] Fix(sorting): absolute sort icon not rendered in column header

sdwvit opened the pull request:
Fix [AG-13490](https://ag-grid.atlassian.net/browse/AG-13490) TC12, TC13, TC14

seanlandsman committed:
DASH-11 Auto grid bump and publish trigger

seanlandsman committed:
DASH-11 Auto grid bump and publish trigger

seanlandsman committed:
DASH-11 Auto grid bump and publish trigger

sdwvit committed:
- remove warning per TC23 in AG-9792
- address TC12 and TC13 in AG-13490

sdwvit committed:
- address TC12 and TC13 in AG-13490

sdwvit committed:
update docs per TC14

sdwvit committed:
Update absolute sorting example and documentation

- Update sortingOrder configuration to match expected behavior for absolute sorting
- Correct sortingOrder array order in absolute sorting example
- Update example to use smaller dataset for better performance
- Improve documentation clarity for absolute sorting feature
- Add explicit mention of sortingOrder in API documentation section

sdwvit committed:
fix lint

StephenCooper commented on documentation/ag-grid-docs/src/content/docs/row-sorting/index.mdoc:
```diff
@@ -127,12 +127,16 @@ The example below shows different combinations of sorting orders as follows:
-## Example: Absolute Value Sorting
-The example below shows how to use a sort definition to allow sorting by absolute value.
+## Absolute Sorting
+Absolute Sorting allows you to sort numeric values based on their magnitude, ignoring their sign. This is useful when you want to rank values by their size rather than by positive or negative direction.
```
```suggestion
Absolute Sorting enables sorting numeric values based on their magnitude, ignoring their sign. This can be used to rank values by their size ignoring if a value is positive or negative.
```

StephenCooper replied:
When writing docs we should try to avoid the words "you" and instead write in the passive voice. Just something to be aware of.

StephenCooper commented on packages/ag-grid-community/src/entities/agColumn.ts:
```diff
@@ -854,3 +868,26 @@ export function _normalizeSortDirection(sortDirectionLike?: unknown): SortDirect
+/**
+ * Returns a flat set of provided menu items names
+ */
+function flattenMenuItems(columnMainMenuItems: (DefaultMenuItem | MenuItemDef)[]) {
```
This code should not be part of agColumn. It should live in the column menu and be accessed via the column menu service. 

This means the code does not bloat core and it is only required when using the column menu.

sdwvit replied:
That was my initial idea. the method is static and does require additional interface for the menu service to be implemented. I'll change it a bit later to be in that module.

StephenCooper committed:
Update documentation/ag-grid-docs/src/content/docs/row-sorting/index.mdoc

SalvatorePreviti commented on packages/ag-grid-community/src/entities/agColumn.ts:
```diff
@@ -451,9 +452,22 @@ export class AgColumn<TValue = any>
+
+        if (Array.isArray(mainMenuItems) && mainMenuItems.length && colMenuFactory) {
+            // eslint-disable-next-line sonarjs/no-ignored-return
+            flattenMenuItems(mainMenuItems).find((k) => {
```
why .find if we don't use the return value? cannot we do a for const of loop?

sdwvit replied:
we can, but it's a bit more elegant to do an early return from a loop

sdwvit replied:
I think you are right actually, this way I don't need to fight with sonar

sdwvit committed:
- Moved flattenMenuItems to IColumnMenuFactory for better structure
 - Updated ColumnMenuFactory to implement IColumnMenuFactory
 - Removed redundant flattenMenuItems function from agColumn.ts
 - Added colMenuFactory to context for improved dependency injection

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
lint

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
extracted static method

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
lint

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
lint

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
pr comment

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
pr review

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
Merge tag 'latest' into sdwvit/fix/sorting/absolute-sort-icon-not-rendered-in-column-header/AG-13490

Latest commit to pass GitHub Actions workflow on latest branch.

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

# Conflicts:
#	.github/workflows/beta-publish.yml

sdwvit committed:
Merge branch 'refs/heads/latest' into sdwvit/fix/sorting/absolute-sort-icon-not-rendered-in-column-header/AG-13490

sdwvit committed:
remove weird file

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>
